### PR TITLE
[improvement](MOW) use seperated cache for mow pk cache

### PR DESCRIPTION
**Pull request by yixiutt**
In mow, primary key cache have a big impact on load performance, so we add a new cache type to seperate 
it from page cache to make it more flexible in some cases

# Proposed changes

Issue Number: close #xxx

## Problem summary

Describe your changes.

## Checklist(Required)

* [ ] Does it affect the original behavior
* [ ] Has unit tests been added
* [ ] Has document been added or modified
* [ ] Does it need to update dependencies
* [ ] Is this PR support rollback (If NO, please explain WHY)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Review by zhannngchen — COMMENTED** [review]
@xinyiZzz pls help to review this patch, do we missed some GC process?

File: be/src/olap/page_cache.cpp [hunk 1]
```diff
@@ -51,6 +52,10 @@ StoragePageCache::StoragePageCache(size_t capacity, int32_t index_cache_percenta
     } else {
         CHECK(false) << "invalid index page cache percentage";
     }
+    if (pk_index_cache_capacity > 0) {
+        _pk_index_page_cache = std::unique_ptr<Cache>(new_lru_cache(
```
Comment (xinyiZzz) [inline be/src/olap/page_cache.cpp:56]: Why not let pk_index_page split a part in page cache, like index cache, now the total cache is too large

**Comment by xinyiZzz** [comment]
We need gc cache when the memory exceed limit, such as releasing all pk index caches, add in MemInfo::process_cache_gc`
```
void MemInfo::process_cache_gc(int64_t& freed_mem) {
     ...
    if (StoragePageCache::instance()->get_page_cache_mem_consumption(segment_v2::PRIMARY_KEY_INDEX_PAGE) >
         min_free_size) {
         freed_mem +=
                 StoragePageCache::instance()->get_page_cache_mem_consumption(segment_v2::PRIMARY_KEY_INDEX_PAGE);
         StoragePageCache::instance()->prune(segment_v2::PRIMARY_KEY_INDEX_PAGE);
     }
}
```
If want to release part of the cache, you need to write Predicate Function, call `StoragePageCache::instance()->prune_if`

File: be/src/olap/page_cache.cpp (hunk 1 above)
Comment (yixiutt) [reply be/src/olap/page_cache.cpp:56]: We want pk index page to be seperated from storage page cache cause storage page cache may not enable by default, but pk index cache may always be enabled.

**Comment by yixiutt** [comment]
> Function

**Comment by yixiutt** [comment]
run buildall

**Comment by yixiutt** [comment]
run buildall

**Review by xinyiZzz — DISMISSED** [review]
LGTM

**Comment by yixiutt** [comment]
run buildall

**Review by xinyiZzz — APPROVED** [review]
LGTM

**Pull request merged**